e2e/tests/rollkit/rollkit_test.go

```diff
@@ -0,0 +1,237 @@
+//go:build !test_e2e
+
+package rollkit
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/cosmos/ibc-go/e2e/dockerutil"
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	apitypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/strangelove-ventures/interchaintest/v8/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	"github.com/strangelove-ventures/interchaintest/v8/testutil"
+	testifysuite "github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+)
+
+const (
+	// these images are being develepod in the the gm-demo repo
+	// https://github.com/chatton/gm-demo
+	rollkitAppRepo    = "ghcr.io/chatton/rollkit"
+	rollkitAppVersion = "latest"
+	wasmSimappRepo    = "ghcr.io/chatton/ibc-go-wasm-simd"
+	wasmSimappVersion = "latest"
+
+	// this image is the one used in the rollkit demo.
+	celestiaImage = "ghcr.io/rollkit/local-celestia-devnet:v0.12.7"
+)
+
+func TestRollkitTestSuite(t *testing.T) {
+	testifysuite.Run(t, new(RollkitTestSuite))
+}
+
+type RollkitTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+func (s *RollkitTestSuite) Test_Rollkit_Succeeds() {
+	cli, networkID, celestiaHostPort := s.createAndStartCelestiaContainer()
+	_, _ = s.SetupChainsRelayerAndChannel(context.TODO(), nil, func(options *testsuite.ChainOptions) {
+		// use existing docker cli and network so the rollkit app can talk to celestia
+		options.DockerNetwork = networkID
+		options.DockerClient = cli
+
+		// TODO: this is not being propagated correctly.
+		// The test will currently fail when the relayer attempts to connect to the rollkit app.
+		options.SkipPathCreation = true
+
+		options.ChainASpec.ChainName = "rollkit"
+		options.ChainASpec.ChainID = "rollkit-app"
+		options.ChainASpec.Bin = "gmd"
+		options.ChainASpec.Bech32Prefix = "gm"
+		options.ChainASpec.AdditionalStartArgs = []string{
+			"--rollkit.aggregator",
+			fmt.Sprintf("--rollkit.da_address=celestia:26650"),
+			fmt.Sprintf("--rollkit.da_start_height=%s", s.getCelestiaBlockHeight(celestiaHostPort)),
+			"--minimum-gas-prices=0.025atoma",
+
+			// NOTE: the following feels shouldn't be required, but the rollkit app fails to communicate with
```

Comment:
ref: https://github.com/strangelove-ventures/interchaintest/pull/952#issuecomment-2010473851 may help here using `HostPortOverride`

Reply:
ah looks perfect, thank you!